### ISSUE TITLE
Cannot restore storage folder

### ISSUE THREAD

**Issue by R4meau** [BoostIO/Boostnote#1319]
So I tried syncing my notes between my mobile app and my desktop app (Linux) and this is when it all broke.

My Boostnote stopped working right after I tried syncing them. I forgot to post an issue about this too. Luckily for me, the Dropbox syncing worked fine, I can see all my `.md` files in my Dropbox. So I re-downloaded Boostnote and tried to link the `boostnote-mobile` folder. To no success.

I opened the developer console and here's the error I'm getting whenever I press the `Create` button after choosing the folder: Uncaught (in promise) TypeError: Cannot read property 'forEach' of undefined

This is getting on my nerves. I have so many notes that I don't want to lose and I love Boostnote. I'd love to contribute and fix this issue but I don't have the time right now.

![screenshot from 2017-12-21 05-29-06](https://user-images.githubusercontent.com/3984265/34252253-db2eb4d4-e610-11e7-9a6d-ddd4980e1dca.png)

**Comment by chiangs** [comment]
Hi nick,

I turned my storage into a git repo to avoid an unforseen disaster. Might
give that a try while things get worked out.

Best regards,

Stephen E.  Chiang

**Comment by R4meau** [comment]
@chiangs Haha, thanks for the suggestion. Will do. If by next week no one fixes it I'll give it a try.

**Comment by float168** [comment]
Hi,
I came across similar error after I created some documents on boostnote-mobile.
In my case, I found just removing a strange cson file which contains only "content" field leads to restore normal state.

I suspect this bug is involved in boostnote-mobile.
Might help you.

**Comment by R4meau** [comment]
@float168 Haha, that's interesting, I did find this file. I deleted it but when I tried to restore my storage again, I got this new error: `Uncaught (in promise) TypeError: Cannot read property 'isTrashed' of undefined`. 

Thanks for the suggestion anyway. At least there's a progress.

**Comment by tasuwo** [comment]
Hi,
I saw the same error, and resolved it.
The error, `Uncaught (in promise) TypeError: Cannot read property 'isTrashed' of undefined`), occurred after updating my Boostnote version from 0.8.5 to 0.10.0.

After occurred error, I found some my notes which doesn't have `isTrashed`.
I think it maybe caused by the cson format changing, in the following commit.

> https://github.com/BoostIO/Boostnote/commit/2650cc2f1c7aeeebf159258fc5d010be5a81852e

To solve this, find the notes which not have `isTrashed` and append the node manually. I realized this by shell script as following.

``` shell
array=`find . -name '*.cson'`

for f in $array; do
    echo $f | xargs grep --quiet 'isTrashed';
    if [ $? = 1 ]; then
      echo 'isTrashed: false' >> $f
    fi
done
```

Execute the script under the directory of notes, add storage again, it works well for me.
